src/components/table/SnapshotAccessRequestTable.tsx

```diff
@@ -58,7 +59,13 @@ function SnapshotAccessRequestTable({
     {
       label: 'Request Name',
       name: 'snapshotName',
-      render: (row: SnapshotAccessRequestResponse) => row.snapshotName || '',
+      render: (row: SnapshotAccessRequestResponse) => (
+        <Modal
```

Comment:
I am not sure `Modal` is the right name for this component. Maybe `ModalTrigger`? The reason I say that is that `Modal` very specifically refers to the dialog that opens on the screen. This also handles the rendering of the text, and open state, so its purview is larger.